Tests/AlicerceTests/AutoLayout/ConstraintGroupToggleTestCase.swift

```diff
@@ -0,0 +1,254 @@
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint1 = view0.bottom(to: host, offset: -100)
+                }
+
+            case .second:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint2 = view0.top(to: host, offset: -100)
+                }
+
+            case .third:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint3 = view0.leading(to: host, offset: -100)
+                }
+            }
+        }
+
+        group.activate(.second)
+        group.activate(.third)
+        group.activate(.first)
+
+        XCTAssertTrue(constraint1.isActive)
+        XCTAssertFalse(constraint2.isActive)
+        XCTAssertFalse(constraint3.isActive)
+    }
+
+    func testReActivateConstraintCurrentConstraint_WithEnumKey_ShouldActivateFirstConstraint() {
+
+        var constraint1: NSLayoutConstraint!
+        var constraint2: NSLayoutConstraint!
+        var constraint3: NSLayoutConstraint!
+
+        let group = ConstraintGroupToggle(initial: TestConstraintGroupKey.first) {
+            [weak self] in
+
+            guard let self = self else { return nil }
+
+            switch $0 {
+            case .first:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint1 = view0.bottom(to: host, offset: -100)
+                }
+
+            case .second:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint2 = view0.top(to: host, offset: -100)
+                }
+
+            case .third:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint3 = view0.leading(to: host, offset: -100)
+                }
+            }
+        }
+
+        group.activate(.first)
+
+        XCTAssertTrue(constraint1.isActive)
+        XCTAssertNil(constraint2)
+        XCTAssertNil(constraint3)
+    }
+
+    func testActivateUnhandledKeyCase_WithNonFiniteKeySet_ShouldMaintainCurrentConstraint() {
+
+        var constraint1: NSLayoutConstraint!
+        var constraint2: NSLayoutConstraint!
+
+        let group = ConstraintGroupToggle(initial: "constraint1") { [weak self]  in
+
+            guard let self = self else { return nil }
+
+            switch $0 {
+            case "constraint1":
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint1 = view0.bottom(to: host, offset: -100)
+                }
+
+            case "constraint2":
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint2 = view0.bottom(to: host, offset: -100)
+                }
+
+            default:
+                return nil
+            }
+
+        }
+
+        group.activate("constraint3")
+
+        XCTAssert(constraint1.isActive)
+        XCTAssertNil(constraint2)
+    }
```

Comment:
should we try to add a test case where the constraint groups are incompatible (thus need to be mutually exclusive), so that we can test that the mechanics are validated for a use case that resembles reality more?